RevenueCatUI/Views/TimelineView.swift

```diff
@@ -0,0 +1,322 @@
+                linkSegments
+            }
+        }
+    }
+    
+    @ViewBuilder
+    var linkSegments: some View {
+        switch configuration.linkPosition {
+        case .leading:
+            linkSegment(configuration.linkColor)
+            linkSegment(.clear)
+        case .trailing:
+            linkSegment(.clear)
+            linkSegment(configuration.linkColor)
+        case .both:
+            linkSegment(configuration.linkColor)
+            linkSegment(configuration.linkColor)
+        }
+    }
+    
+    @ViewBuilder
+    func linkSegment(_ color: Color = .clear) -> some View {
+        if axis == .vertical {
+            Rectangle()
+                .fill(color)
+                .frame(width: .infinity, height: 2)
+        } else {
+            Rectangle()
+                .fill(color)
+                .frame(width: 2, height: .infinity)
+        }
+    }
+    
+    @ViewBuilder
+    var content: some View {
+        if axis == .vertical {
+            VStack(spacing: 12) {
+                Text(configuration.title)
+                    .font(.system(size: 13))
+                    .bold()
+                iconWithBackground
+                Text(configuration.subtitle)
+                    .foregroundStyle(.secondary)
+                    .font(.system(size: 11))
+            }
+        } else {
+            HStack(alignment: .top, spacing: 12) {
+                iconWithBackground
+                    .alignmentGuide(.iconAlignment) {
+                        $0[HorizontalAlignment.center]
+                    }
+                VStack(alignment: .leading, spacing: 2) {
+                    Text(configuration.title)
+                        .font(.system(size: 17))
+                        .bold()
+                    Text(configuration.subtitle)
+                        .foregroundStyle(.secondary)
+                        .font(.system(size: 12))
+                }
+            }
+        }
+    }
+    
+    @ViewBuilder
+    var iconWithBackground: some View {
+        ZStack {
+            Circle()
+                .fill(configuration.iconBackgroundColor)
+                .overlay(
+                    RoundedRectangle(cornerRadius: iconSize / 2)
+                        .stroke(TimelineView.Colors.link, lineWidth: 1)
+                )
+            Image(systemName: configuration.icon)
+                .resizable()
+                .scaledToFit()
+                .frame(width: iconSize / 2, height: iconSize / 2)
+                .foregroundColor(configuration.iconForegroundColor)
+        }
+        .frame(width: iconSize, height: iconSize)
+    }
+}
+
+#if DEBUG
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+struct Preview: View {
+    
+    @Environment(\.horizontalSizeClass)
+    var horizontalSizeClass
+    
+    var body: some View {
+        if horizontalSizeClass == .regular {
+            TimelineView(stepConfigurations: [
```

Comment:
can you render both variants at the same time instead? faster to iterate/more visual.